wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -106,45 +106,47 @@ private[wallet] trait UtxoHandling extends WalletLogger {
           case None =>
             logger.warn(
               s"Given txos exist in block (${blockHash.hex}) that we do not have! $txos")
-            txos
+            Vector.empty
           case Some(confs) =>
-            txos.map { txo =>
+            txos.flatMap { txo =>
               txo.state match {
                 case TxoState.ImmatureCoinbase =>
                   if (confs > Consensus.coinbaseMaturity) {
                     if (confs >= walletConfig.requiredConfirmations) {
-                      txo.copyWithState(TxoState.ConfirmedReceived)
+                      Some(txo.copyWithState(TxoState.ConfirmedReceived))
                     } else {
-                      txo.copyWithState(TxoState.PendingConfirmationsReceived)
+                      Some(
+                        txo.copyWithState(
+                          TxoState.PendingConfirmationsReceived))
                     }
                   } else {
-                    txo
+                    None
```

Comment:
Don't  we need some sort of logic here to handle the reorg case? I.e. if we had pending confirmations, and then a reorg happened, we need to go back to immature coinbase?

Reply:
A shorter chain won't cause reorg so it's not possible for a reorg to bring us back to immature. However, it is possible that our block is reorged and our immature coinbase will no longer be part of the chain. We should be safe from this unless there is over a 101 block reorg because if our coinbase was reorged at say 3 confs it would just stay stuck as `ImmatureCoinbase`